src/types.ts

```diff
@@ -1,33 +1,29 @@
-// Human-readable generic types
-type Id<T> = T
-
 // Extract the argument/return types of a function
-type In<T> = T extends (...args: infer U) => any ? U : []
-type Out<T> = T extends (...args: any[]) => infer U ? U : never
-
-// Extract keys whose values match a condition
-type Filter<T, Cond, U extends keyof T = keyof T> = {
-  [K in U]: T[K] extends Cond ? K : never
-}[U]
+type In<T> = T extends (...args: infer U) => any ? U : unknown[]
 
-// Extract an array type of valid event keys
-export type EventKey<T> = Filter<T, (...args: any[]) => any> & string
+export type Falsy = false | null | undefined
 
-// Extract the argument/return types of a valid event
-export type EventIn<T, K extends EventKey<T>> = Id<In<T[K]>>
-export type EventOut<T, K extends EventKey<T>> = Id<Out<T[K]> | void>
+/** Extract an array type of valid event keys */
+export type EventKey<T> = keyof T & string
```

Comment:
qq: Are you removing support for output values in v2?

Reply:
Sort of. You'll still be able to use `getListeners` with `Array.from` like this:
```js
const results = Array.from(ee.getListeners('key'), fn => fn())
```